specs/kvstore/WriterFencing.fizz

```diff
@@ -0,0 +1,150 @@
+---
+action_options:
+    SlateDBWriter.FlushWal:
+        max_actions: 3
+---
+
+# Model to validate that there can only be one writer and that old writers can be fenced.
+SUCCESS = "success"
+CONFLICT = "conflict"
+NUM_WRITERS = 2
+
+role ObjectStore:
+  action Init:
+    self.objects = {}
+
+  atomic func ConditionalWrite(name, content):
+    if name in self.objects.keys():
+        return CONFLICT
+    self.objects[name] = content
+    return SUCCESS
+
+  atomic func Read(name):
+    return self.objects.get(name)
+
+  atomic func List():
+    return sorted(self.objects.keys())
+
+symmetric role SlateDBWriter:
+    action Init:
+        self.state = "waiting"
+        self.flush_count = 0
+        self.was_ever_open = 0
+    
+    atomic action FlushWal:
+        if self.state != "opened":
+            return
+        self.WriteWal()
+        self.flush_count += 1
+    
+    func LoadLatestManifest():
+        manifests = manifest_store.List()
+        result = CONFLICT
+        if len(manifests) > 0:
+            recent_manifest = manifest_store.Read(manifests[len(manifests) - 1])
+            wal = wal_store.List()
+            self.next_wal = wal[-1] + 1 if wal else 1
+            result = self.WriteManifest(record(id = recent_manifest.id + 1, writer_epoch = recent_manifest.writer_epoch + 1))
+        else:
+            atomic:
```

Comment:
Is this required to be atomic? Removing `atomic` also works. Since writing to object store and updating next_wal locally cannot be truly atomic, it probably can be removed safely. What do you think?